### ISSUE TITLE
Intermittent pass in css21_dev/html4/quotes-036.htm

### ISSUE THREAD

**Issue by jdm** [servo/servo#7846]
```
 3:24.10 TEST_START: Thread-TestrunnerManager-33 /css21_dev/html4/quotes-036.htm
 3:25.00 TEST_END: Thread-TestrunnerManager-33 PASS, expected FAIL

/css21_dev/html4/quotes-036.htm
-------------------------------
PASS expected FAIL [Parent]
```

**Comment by notriddle** [comment]
Isn't it weird how this test fails in the exact same way in FF and Chrome?

![image](https://cloud.githubusercontent.com/assets/1593513/10385660/e6ce069e-6e00-11e5-9b89-46ea23259459.png)

**Comment by Ms2ger** [comment]
A little, but this is one of Hixie's evil tests, after all. It's more surprising that we sometimes pass it.

**Comment by notriddle** [comment]
69db83476d02d06658ad084a834abc94f7f008d0 is when it started intermittently passing. It's intermittent because it's not doing style recalc when it (presumably) should. Notice that if you open the test case manually, it'll switch from fail mode to pass mode if you hover it.

**Comment by Ms2ger** [comment]
@mbrubeck want to debug this one?

**Comment by nox** [comment]
Interestingly, disabling incremental layout produces a blank page.

**Comment by mbrubeck** [comment]
Servo lays out this test correctly in its initial layout, but then messes it up in any relayout (whether it's an incremental or full layout).

The problem is that the ResolveGeneratedContent traversal is not safe to run more than once on the same flow.  It mutates some GeneratedContent fragments into ScannedText fragments. The next time it runs, any remaining GeneratedContent fragments are processed again but with an incorrect correct quote nesting level (because some of the surrounding GeneratedContent fragments are gone).

> Interestingly, disabling incremental layout produces a blank page.

This is fixed by #9964.
